Remove needsInputModeNoneImplicitly

LumioseSil opened the pull request:
#### 87a1ee7dd4f2231b246ea65f1ef7ac0b8ac0d06a
<pre>
Remove needsInputModeNoneImplicitly
<a href="https://bugs.webkit.org/show_bug.cgi?id=252819">https://bugs.webkit.org/show_bug.cgi?id=252819</a>

Reviewed by Ryosuke Niwa.

After checking desmos.com and viewing its source, I can confirm the site
now adopts inputmode=&quot;none&quot;, so we can remove this quirk.

*Source\WebCore\html\HTMLElement.cpp: Remove
needsInputModeNoneImplicitly check

*Source\WebCore\page\Quirks.cpp: Remove needsInputModeNoneImplicitly
*Source\WebCore\page\Quirks.h: Remove needsInputModeNoneImplicitly

Canonical link: <a href="https://commits.webkit.org/261175@main">https://commits.webkit.org/261175@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/357f6e788a84ba03fdfe8ab9fcf5c1c8216f6f39

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109345 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18420 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42146 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/878 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118545 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/113228 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19969 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9698 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/431 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115101 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14885 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/98112 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101578 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96857 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29765 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43071 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11244 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31108 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84823 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11919 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/8039 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17286 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50711 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7745 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13544 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (dismissed):
Marking as "Request changes" until Karl has verified.

LumioseSil committed:
Remove needsInputModeNoneImplicitly
https://bugs.webkit.org/show_bug.cgi?id=252819

Reviewed by Ryosuke Niwa.

After checking desmos.com and viewing its source, I can confirm the site
now adopts inputmode="none", so we can remove this quirk.

*Source\WebCore\html\HTMLElement.cpp: Remove
needsInputModeNoneImplicitly check

*Source\WebCore\page\Quirks.cpp: Remove needsInputModeNoneImplicitly
*Source\WebCore\page\Quirks.h: Remove needsInputModeNoneImplicitly

Canonical link: https://commits.webkit.org/261175@main
